### DISCUSSION TITLE
Decodes GUI running in the cloud

### DISCUSSION THREAD

**Discussion by ktarbet** [opendcs/opendcs#340] [General]
I'm testing OpenDCS inside AWS-AppStream; Using a Windows Virtual Environment, where there is just OpenDCS installed, nothing else.   

By default the environment is restrictive in the install directory that I chose:   c:\apps\opendcs  

Running launcher_start.bat gives this error:

```
IOException trying to open log file 'gui.log': java.io.FileNotFoundException: gui.log (Access is denied)

```

Any suggestions?
Can the log location be configured?

**Comment by MikeNeilson** [comment]
you can pass `-l <filename` (that's a lower case L. However, the launcher will open app logs in the default location.

You'll want to look at the integration tests and how it set's up the DCSTOOL_USERDIR directory. I was going to point you to the docs but I just looked at them and it's about as clear as mud on how to do that.

but the short version is:
environment variables:
DCSTOOL_HOME -> install directory
DECODES_INSTALL_DIR -> install directory
DCSTOOL_USERDIR -> somewhere user writable

And sanity check the .bat files actually set "-DDCSTOOL_USERDIR"

I know i've used it on windows but vaguely remember some issues.

then instead of decodes.properties it's user.properties for the default profile.
(actually make both for the launcher, will be fixing that for the next release.)

**Comment by ktarbet** [comment]
Update on testing AppStream with OpenDCS:

I built an AppStream 2.0 Image, where I installed Java, then OpenDCS, and then configured the AppStream Template user and setup the proper shortcuts.  One challenge was to set the working directory for OpenDCS so there were sufficient permissions to write log files, when testing the application as non-administrator.  I used the AppStream Image Builder that has you switch between Administrator, Test User, and Template User.  

Next Step:

There is a newer feature that allows creating an Elastic Fleet (with a App Block) that I plan to investigate next.  The Elastic fleet may be a better fit for OpenDCS.  The advantage is the apps are installed on-the-fly from a S3 bucket, so versioning may be easier, and elastic Fleets appear more cost effective than  On-Demand fleets.  The elastic Fleet appears best for apps that are around 2Gb or less. 

Here is a quick summary of fleet types from the AWS console:

Always-On
Select this fleet type to provide users instant-on access to their apps. You will be charged for all running instances in your fleet even if no users are streaming apps.

On-Demand
Select this fleet type to optimize your streaming costs. With an on-demand fleet, users will experience a start time of about one to two minutes for their session. However, you will only be charged the streaming instance fees when users are connected, and a small hourly fee for each instance in the fleet that is not streaming apps.

Elastic
Select an elastic fleet for applications that don’t require installation and can be run from a virtual hard disk. Elastic fleets don’t support AppStream 2.0 images, nor do they require scaling policies. You are charged only for the duration of a streaming session.

**Comment by MikeNeilson** [comment]
I didn't actually try to run a specific app yet but this did startup without issue locally: https://docs.linuxserver.io/images/docker-webtop

Shows some promise though is Linux only.

**Comment by ktarbet** [comment]
Here is the proof-of-concept setup for Elastic AppStream

https://github.com/ktarbet/learning/blob/master/aws-appstream.md

**Comment by agilmore2** [comment]
That's awesome work! I'll have to check this out.

**Comment by MikeNeilson** [comment]
Our cloud provider never let us handle it in the way Karl tried above. We've also since started the RestAPI and Web GUI.
